packages/netsuite-adapter/src/query.ts

```diff
@@ -353,6 +369,15 @@ export function validatePlainObject(
   }
 }
 
+export function validateDefined(
+  value: unknown,
+  configPath: string | string[]
+): asserts value is Record<string, unknown> {
+  if (value === undefined) {
+    throw new Error(`${makeArray(configPath).join('.')} should be defined`)
+  }
+}
```

Comment:
This API feels a bit weird to me.
Maybe something in the sense of the below will be clearer?
```suggestion
export function validateDefined(
  value: T | undefined,
  configPath: string | string[]
): asserts value is T {
  if (value === undefined) {
    throw new Error(`${makeArray(configPath).join('.')} should be defined`)
  }
}
```